component/valves/src/main/java/org/wso2/carbon/identity/authenticator/x509cert/valve/X509CertificateAuthenticationValve.java

```diff
@@ -102,4 +103,11 @@ private X509Certificate getCertificate(Request request) {
         }
         return certificate;
     }
+
+    private String getX509HeaderName() {
+
+        config = X509ServerConfiguration.getInstance();
```

Comment:
We can use `IdentityUtil.getProperty("X509. X509RequestHeaderName")` to read
<X509>
   <X509RequestHeaderName>somevalue</X509RequestHeaderName>
</X509>
in identity.xml

Reply:
Since this is not an osgi bundle, lets continue to use `IdentityConfigParser`.

Reply:
auth-rest-valve also use this.
https://github.com/wso2-extensions/identity-carbon-auth-rest/blob/master/components/org.wso2.carbon.identity.auth.service/src/main/java/org/wso2/carbon/identity/auth/service/util/AuthConfigurationUtil.java#L62